chapters/cls-metrics.qmd

```diff
@@ -0,0 +1,1693 @@
+
+For the event class, Q2 is measured by a statistic called the _positive predictive value_ (PPV), which requires sensitivity, specificity, and prevalence: 
+
+$$
+PPV= \frac{Prev \times Sens}{(Sens \times Prev) + \left((1 - Spec) \times (1 - Prev)\right)}
+$$  {#eq-ppv}
+
+Without knowing anything about how often the average mushroom is poisonous, we could choose a prevalence of 50{{< pct >}} as an uninformative prior. Our estimate from the training set (`r obs_prev`{{< pct >}}) is close to uninformative so that we will use that in our computations. 
+
+With our estimates in hand, the answer to Q2 is 
+
+> A2: `r round(nb_mtr$mean[nb_mtr$.metric == "ppv"] * 100, 1)`{{< pct >}} of the time, a mushroom is poisonous when we predict it to be."
+
+The Bayesian analog to specificity is called the _negative predictive value_ (NPV) and has a similar formula: 
+
+$$
+NPV= \frac{(1 - Prev) \times Spec}{\left(Prev \times (1 - Sens)\right) + \left(Spec \times (1 - Prev)\right)}
+$$  {#eq-npv}
+
+Using the prevalence from our training set and the metrics computed on the validation set, the NPV estimate is `r round(nb_mtr$mean[nb_mtr$.metric == "npv"] * 100, 1)`{{< pct >}}. 
+
+The PPV and NPV are functions of three quantities, and it might be challenging to understand how the prevalence can affect the statistics. @fig-prevalence can help us understand these relationships. For example, if the prevalence is near zero, achieving an acceptable PPV value may be difficult unless the specificity is perfect (in which case it is quite easy to attain a good NPV). 
+
+::: {#fig-prevalence}
+
+::: {.figure-content}
+
+```{shinylive-r}
+#| label: fig-prevalence
+#| viewerHeight: 550
+#| standalone: true
+
+library(shiny)
+library(dplyr)
+library(purrr)
+library(ggplot2)
+library(patchwork)
+library(bslib)
+
+source("https://raw.githubusercontent.com/aml4td/website/main/R/shiny-setup.R")
+
+# ------------------------------------------------------------------------------
+
+ui <- fluidPage(
+#  theme = grid_theme,
+  layout_columns(
+    fill = FALSE,
+    col_widths = breakpoints(sm = c(-3, 6, -3)),
+    column(
+      width = 10,
+      sliderInput(
+        inputId = "prev",
+        label = "Prevelance",
+        min = 0,
+        max = 1,
+        value = 1/2,
+        width = "100%",
+        step = 1/20
+      )
+    )
+  ),
+  as_fill_carrier(plotOutput('results'))
+)
+
+server <- function(input, output) {
+  
+  ppv_fn <- function(sens, spec, prev = 1 / 2) {
+    ( sens * prev ) / ( (sens * prev) + ( (1 - spec) * (1 - prev) ) )
+  }
+  npv_fn <- function(sens, spec, prev = 1 / 2) {
+    ( spec * (1 - prev) ) / ( ( (1 - sens) * prev) + ( (spec) * (1 - prev) ) )
+  }
+  
+  output$results <-
+    renderPlot({
+      
+      x <- seq(0, 1, length.out = 50)
```

Comment:
I'm not completely sold on this way to visualize the results. I'm 100% open to some other visualization of these patterns.